vaadin-checkbox-flow-demo/src/main/java/com/vaadin/flow/component/checkbox/demo/CheckboxGroupView.java

```diff
@@ -34,174 +46,219 @@
+            }
+        });
         // end-source-example
 
-        group.addValueChangeListener(
-                event -> valueInfo.setText(toString(group.getValue())));
-
-        group.setId("checkbox-group-disabled-items");
-        valueInfo.setId("checkbox-group-disabled-items-info");
-
-        addCard("Checkbox group with item enabled provider", group, valueInfo);
+        addCard("Value change event", checkboxGroup, value);
     }
 
-    private void addReadOnlyGroup() {
+    private void indeterminateCheckbox() {
         // begin-source-example
-        // source-example-heading: Read-only checkbox group
-        Div valueInfo = new Div();
+        // source-example-heading: Indeterminate checkbox
+        Checkbox checkbox = new Checkbox("Select all");
+        CheckboxGroup<String> checkboxGroup = new CheckboxGroup<>();
+        Set<String> items = new LinkedHashSet<>(
+                Arrays.asList("Option one", "Option two"));
+        checkboxGroup.setItems(items);
+        checkboxGroup.addThemeVariants(CheckboxGroupVariant.LUMO_VERTICAL);
+        checkboxGroup.addValueChangeListener(event -> {
+            if (event.getValue().size() == items.size()) {
+                checkbox.setValue(true);
+                checkbox.setIndeterminate(false);
+            } else if (event.getValue().size() == 0) {
+                checkbox.setValue(false);
+                checkbox.setIndeterminate(false);
+            } else
+                checkbox.setIndeterminate(true);
+
+        });
+        checkbox.addValueChangeListener(event -> {
+
+            if (checkbox.getValue()) {
+                checkboxGroup.setValue(items);
+            } else {
+                checkboxGroup.deselectAll();
+            }
+        });
+        checkboxGroup.setValue(Collections.singleton("Option one"));
+        // end-source-example
 
-        CheckboxGroup<String> group = new CheckboxGroup<>();
-        group.setItems("foo", "bar", "baz");
-        group.setReadOnly(true);
+        addCard("Indeterminate checkbox", checkbox, checkboxGroup);
+    }
 
-        NativeButton button = new NativeButton("Switch read-only state",
-                event -> group.setReadOnly(!group.isReadOnly()));
-        group.addValueChangeListener(
-                event -> valueInfo.setText(toString(group.getValue())));
+    private void configurationForReqired() {
+        // begin-source-example
+        // source-example-heading: Required
+        Employee employee = new Employee();
+        Binder<Employee> binder = new Binder<>();
+
+        CheckboxGroup<String> checkboxGroup = new CheckboxGroup<>();
+        checkboxGroup.setLabel("Employee title");
+        checkboxGroup.setItems("Account Manager", "Designer",
+                "Marketing Manager", "Developer");
+        checkboxGroup.addThemeVariants(CheckboxGroupVariant.LUMO_VERTICAL);
+
+        binder.forField(checkboxGroup)
+                .asRequired("Please choose an employee title")
```

Comment:
Is this demo copied from some other component's validation example? A lot of the demo source indicates that the user is doing a single selection for an employee's "title" ("Employee title", "Please choose an employee title", "setTitle"). Even the Bean has a field "title" but it's actually a `Set<String>` and this isn't visible in the demo source.

Reply:
Done. (I'm going to talk about this fix with Jouni)